nomad/api/job.py

```diff
@@ -271,16 +271,23 @@ def stable_job(self, id, version, stable):
                        "Stable": stable}
         return self.request(id, "stable", json=revert_json, method="post").json()
 
-    def deregister_job(self, id):
+    def deregister_job(self, id, purge=None):
         """ Deregisters a job, and stops all allocations part of it.
 
            https://www.nomadproject.io/docs/http/job.html
 
             arguments:
               - id
+              - purge (bool), optionally specifies whether the job should be
+                stopped and purged immediately (`purge=True`) or deferred to the
+                Nomad garbage collector (`purge=False`).
+
             returns: dict
             raises:
               - nomad.api.exceptions.BaseNomadException
               - nomad.api.exceptions.URLNotFoundNomadException
         """
-        return self.request(id, method="delete").json()
+        params = None
+        if purge is not None:
```

Comment:
Maybe we can use something instead of separating the checks in the conditional and elvis:
```
if purge is not None and isinstance(purge, bool):
    params = {"purge": purge}
```

Reply:
I'm not sure I understand. 
Is the issue that I use "truthy"/"falsy" instead of type checking purge? I can fix that, but then I'd rather raise a `TypeError` than quietly ignore `purge`. 

Something like this?

```python
params = None
if purge is not None:
    if not isinstance(purge, bool):
        raise TypeError(...)
    # Needs to be lower case, {"purge": purge}  doesn't work.
    params = {"purge": "true" if purge else "false"}
```

Reply:
I agree it shouldn't silently fail and we do have an exception class for [InvalidParemeters](https://github.com/jrxFive/python-nomad/blob/master/nomad/api/exceptions.py#L25). So maybe we can raise that instead.

I also tried verifying that the purge value needs to be lower-case I don't believe that is the case. Against nomad (0.8.5 maybe this is different in previous versions) and the below seems to verify that.

https://github.com/hashicorp/nomad/blob/07bfc856cf765f7720edbe0e4d6b46aa588dd1cf/command/agent/job_endpoint.go#L400-L411

[strconv.ParseBool](https://golang.org/pkg/strconv/#ParseBool) handles several representations of an boolean.

Thanks again for pointing out it shouldn't silently ignore!

Reply:
You're right. Will also raise `InvalidParameters` instead.